### ISSUE TITLE
[HOLD #12603][Image][Performance] Avatars appear grey as they're loaded from server for common mobile flows - reported by @sobitneupane

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#11425]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the mobile app
2. Click `Request Money` from Global Create (green + button)
3. Enter any amount  
4. On the next screen, notice how the color of the avatar for recent contacts starts grey, and as we re-render, the actual avatars/list appears.

Note: If you check the [linked conversation](https://expensify.slack.com/archives/C01GTK53T8Q/p1663403937756359), then it's also easy to see that this happens in the IOU preview component. You can also reproduce this on most mobile flows. I was able to do the same with `Send Money` as well as `New Chat`.

## Expected Result:
Avatars are saved locally and don't re-render for our most common mobile flows, let alone your most frequently used contact.

## Actual Result:
Recent avatars start out as grey since that's all we have locally, and after being downloaded from the server, the actual default avatar appears

## Workaround:
None

## Platform:

Where is this issue occurring?

- iOS
- mWeb
- Web

**Version Number:** 1.2.9-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** 
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/1127863/193144887-7a87a5d9-fb2f-4080-8513-10e85fd41e46.mov

**Expensify/Expensify Issue URL:** 
**Issue reported by:**  @sobitneupane 
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1663403937756359

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by JmillsExpensify** [comment]
I can reproduce this one, but I'm on the fence whether we have anything to do here. Floating internally and then I'll circle back.

**Comment by JmillsExpensify** [comment]
Hmm, these reproduction steps are pretty hard to follow. I'm not seeing any greyness in the video. In fact the original report was a different flow. It was appearing in the IOU preview component.

**Comment by JmillsExpensify** [comment]
That said, I would imagine that the reason the greyness is happening in the `Request Money` flow as well as the IOU preview component because we are re-rendering the avatars. They grey is the skeleton UI color so that's what we fall back on.

**Comment by JmillsExpensify** [comment]
I'm going to update the video with one of my own so that the greyness is visible and more clear.

**Comment by JmillsExpensify** [comment]
Likely related: https://github.com/Expensify/App/issues/6527

**Comment by JmillsExpensify** [comment]
Ok, I also discovered another way to reproduce this issue on Web (as well as mWeb). Reproduction steps for web

1. Open up a chat that has quite a bit of history
2. Start scrolling up quickly
3. At a certain point, you'll see the same issue. Avatars appear a dark gray, at which point they are eventually loaded. 

I think this is a backend/performance issue, so I'm going to add the engineering label for more thoughts.

**Comment by PauloGasparSv** [comment]
Hey @JmillsExpensify I took a quick look but I'll reassign the engineering label so someone else can take it because I have a full plate right now : /

Adding what I found so far here:
___

While investigating this I compared the network tab between NewDot and Whatsapp.
Here is how our chat loads thumbnail images against how Whatsapp loads some images:

**NewDot**
<img width="621" alt="image" src="https://user-images.githubusercontent.com/6564265/193682789-c36b8447-fcc2-4a65-b3bf-02fac5b36680.png">
    
**Whatsapp**
<img width="610" alt="image" src="https://user-images.githubusercontent.com/6564265/193683360-247bdb48-4ccf-4097-ac18-14d42cc98a47.png">

    
As you can see in the Size tab of the screenshots , if an image was already requested in Whatsapp the browser caches it.
    **I think our App should be doing that** and it would probably fix the problem everywhere. But right now we ask the browser not to cache: 

  <img width="693" alt="image" src="https://user-images.githubusercontent.com/6564265/193680694-a04dfc52-e6d7-410b-ad09-92f73b556551.png">

I found this [slack thread](https://expensify.slack.com/archives/C011W8BJ9L6/p1613444195333100) that mentions @thienlnam so maybe he has some info on this!

**Comment by aldo-expensify** [comment]
@PauloGasparSv I think the `cache-control: no-cache` header is added when you have the checkbox `Disable cache` checked in the network tab of chrome dev tools. If I uncheck it, I don't see that header anymore. Having said that, I also think there is something preventing the avatar images from being cached

Headers without the `Disable cache`:

![image](https://user-images.githubusercontent.com/87341702/193692773-dd81b201-8804-4476-a85c-f809f1c43b8d.png)

**Comment by PauloGasparSv** [comment]
🤦‍♂️ thanks @aldo-expensify! 100% forgot about that!

**Comment by JmillsExpensify** [comment]
Going to float this one in open-source because it's nice polish.

**Comment by Beamanator** [comment]
Added this to our Image-related features & improvements tracking issue: https://github.com/Expensify/App/issues/10894

**Comment by parasharrajat** [comment]
FYI, I think this issue will be solved as a co-product of another issue. I see Alex is tracking those together which will help identify such issues.

**Comment by Beamanator** [comment]
Agreed @parasharrajat - this issue is pretty common & the upcoming P/S will aim to solve all of these related "gray avatar" issues

**Comment by Beamanator** [comment]
Changing to Weekly since this is currently on hold

**Comment by JmillsExpensify** [comment]
Great call. Thanks

**Comment by JmillsExpensify** [comment]
Still on hold.

**Comment by mallenexpensify** [comment]
@JmillsExpensify Snaggin' from you cuz I got the tracking issue assigned to me
[Tracking] [Polish] Image-related feature request and improvements #10894

Also assigning @Beamanator cuz he's leading that.

**Comment by mallenexpensify** [comment]
on  hold

**Comment by trjExpensify** [comment]
On hold, Melv.

**Comment by trjExpensify** [comment]
@Beamanator, shall we more specifically note this one as held on caching or am I off the boil with that one?

**Comment by Beamanator** [comment]
That's a good idea @trjExpensify - you're on the right track! Ideally once caching works well, this issue will just "be fixed"

**Comment by Beamanator** [comment]
But this also may depend on @roryabraham 's default avatar work actually 👍 I think we have a specific issue for that? Or am i confused?

**Comment by trjExpensify** [comment]
What is this illusive issue you keep referencing? 😅

**Comment by Beamanator** [comment]
Ok ok ok so it's related to this: https://github.com/Expensify/App/issues/9703

But we don't have a specific issue for it "yet" - I am hoping that issue will turn into the default-hard-coded-avatar-implementation :D

**Comment by trjExpensify** [comment]
Got it, nice. Okay! So, @cristipaval is taking that initiative on in the other issue. I'm going to put a hold in the title for the remaining caching issue on web/desktop. After which, we can retest and if it still doesn't work, we can update to #9703. Sound good?

**Comment by Beamanator** [comment]
That sounds great 👍

**Comment by Beamanator** [comment]
Still on hold 🙃

**Comment by trjExpensify** [comment]
Same as [here](https://github.com/Expensify/App/issues/11425#issuecomment-1326593291), Melv. Held on the caching issues.

**Comment by mallenexpensify** [comment]
Still on hold, removing my assignment cuz TRJ is managing the tracking issues and all related

**Comment by Beamanator** [comment]
Still on hold, main caching issues are making progress so hopefully this will come off hold in the next few weeks

**Comment by trjExpensify** [comment]
Still held, but we're [close](https://github.com/Expensify/App/issues/12603#issuecomment-1376125732).

**Comment by JmillsExpensify** [comment]
This one hit me several times today! Can't wait.

**Comment by Beamanator** [comment]
Still held

**Comment by Beamanator** [comment]
Still held

**Comment by trjExpensify** [comment]
Still held on the caching App PR

**Comment by trjExpensify** [comment]
Still held on the caching PR which is being held by a CORS issue.

**Comment by Beamanator** [comment]
Still held

**Comment by trjExpensify** [comment]
Agreed. No change here on the status of being held, for Melv!

**Comment by trjExpensify** [comment]
No change, Melv. I've commented on #12603 to get that updated to the wider project it's now being held on.

**Comment by trjExpensify** [comment]
@justinpersaud is there a tracking issue for the server migration yet?

**Comment by justinpersaud** [comment]
Not yet. I've been tied up closign off loose ends in the #keep-our-banking-connections project and other issues I've had lingering. Planning to get that moving start of next week. Maybe I can just create the GH for it now though and we can take it from there.

**Comment by justinpersaud** [comment]
https://github.com/Expensify/Expensify/issues/277714

**Comment by Beamanator** [comment]
Still held

**Comment by trjExpensify** [comment]
Dropping to monthly, as what's left here for the image improvements project is held on caching > which is blocked on the server migration project.

**Comment by trjExpensify** [comment]
Same as above. What's left here for the image improvements project is held on caching > which is blocked on the server migration project in progress.

**Comment by trjExpensify** [comment]
No change, Melv.

**Comment by Beamanator** [comment]
same

**Comment by trjExpensify** [comment]
Same as above. What's left here for the image improvements project is held on caching > which is blocked on the server migration project in progress.

**Comment by Beamanator** [comment]
stilllll held

**Comment by trjExpensify** [comment]
CORS issue complete, web image caching in review. Getting closer!

**Comment by trjExpensify** [comment]
Image caching PR was deployed and reverted. Awaiting this now before re-implementing: https://github.com/Expensify/App/issues/32703

**Comment by blimpich** [comment]
https://github.com/Expensify/App/issues/32703 is fixed, good to move forward with this.

**Comment by trjExpensify** [comment]
Caching PR was reverted again, [new PR](https://github.com/Expensify/App/pull/36560) is in the works.

**Comment by trjExpensify** [comment]
Alrighty, that PR has been deployed and doesn't look like it'll get reverted. We need to work on the new cache API next, but I believe that's in a line of incremental improvements in vip-vsb. That said, I think we somewhat solved this "grey avatar" problem in the OP by using the skeleton UI on the participant selector when it's loading. So while I think improved caching could maybe be of help here for the avatars, I don't think we need to pursue this issue standalone. With that, I'm inclined to keep it off VIP/Wave and close it now. @Beamanator, what do you think?

**Comment by Beamanator** [comment]
That definitely makes sense to me @trjExpensify , let's close this bad boi!

**Comment by trjExpensify** [comment]
Nice!
